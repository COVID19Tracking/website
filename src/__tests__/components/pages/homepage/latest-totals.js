import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import LatestTotals from '~components/pages/homepage/latest-totals'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    covidUs: {
      totalTestResults: 5795728,
      positive: 1005592,
      death: 52525,
    },
  }))
})

describe('Components : Pages : Homepage : Latest Totals', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LatestTotals />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
