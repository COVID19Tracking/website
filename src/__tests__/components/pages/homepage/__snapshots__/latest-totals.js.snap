// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Homepage : Latest Totals renders correctly 1`] = `
<div
  className="container"
>
  <p
    className="homepage-paragraph paragraph"
  >
    Every day, our volunteers compile the latest numbers on tests, cases, hospitalizations, and patient outcomes from every US state and territory.
  </p>
  <div
    className="wrapper"
  >
    <span
      className="ctaLink"
    >
      <a
        className="cta"
        href="/data"
      >
        See all data
        <span
          aria-hidden={true}
          className="arrow"
        >
           
          â†’
        </span>
      </a>
    </span>
    <div
      className="headerContainer"
    >
      <h2>
        Latest totals:
      </h2>
    </div>
    <div
      className="row allTotals"
    >
      <div
        className="col colSmall4 colMedium6 colLarge4"
      >
        <div
          className="total"
          role="text"
        >
          <div
            className="number"
          >
            5,795,728
          </div>
          <div
            className="label"
          >
            Total test results
          </div>
        </div>
      </div>
      <div
        className="col colSmall4 colMedium6 colLarge4"
      >
        <div
          className="total"
          role="text"
        >
          <div
            className="number"
          >
            1,005,592
          </div>
          <div
            className="label"
          >
            Cases
          </div>
        </div>
      </div>
      <div
        className="col colSmall4 colMedium6 colLarge4"
      >
        <div
          className="total"
          role="text"
        >
          <div
            className="number"
          >
            52,525
          </div>
          <div
            className="label"
          >
            Deaths
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
